src/main/java/reactor/netty/http/client/HttpClient.java

```diff
@@ -96,10 +88,9 @@
  * @author Stephane Maldini
  * @author Violeta Georgieva
  */
-public abstract class HttpClient {
+public abstract class HttpClient extends TransportClient<HttpClient, HttpClientConfig> {
```

Comment:
It is now possible to call `httpClient.connect()` it seems which doesn't make sense. I can suggest a separate interface `ConnectableTransport` interface that `HttpClient` does not implement.

Related to that, `HttpClientConnect` doesn't need anything from `HttpClient` and I don't see why it extends it, and likewise `HttpClient` could be self-sufficient, and only needs its own `config` field.

So then `HttpClient` could extend `ClientTransport` (without any `connect` methods), while `HttpClientConnect` doesn't extend anything and only implements `ConnectableTransport`.

If we do that then the names `HttpClientConnect`, `HttpClientFinalizer`, and `WebSocketFinalizer` no longer make sense. They could be `HttpConnectableTransport`, `HttpRequestSender`, and `WebSocketRequestSender`.

Reply:
@rstoyanchev I like the thinking here I wondered as much and we discussed that over a call with @violetagg .  The intent was to avoid multiplying the contracts but I can see how it deteriorates readability in that case.

An alternative would be to keep that connect() to provide an `HttpConnection` (duplex) with sendRequest/receiveResponse methods. The way `WebClient` works would easily align with this and gives an easier lifecycle management for users building on top of the client where users would still prefer the guided api (and the finalizers should change name as you suggested).

Reply:
in addition to the @smaldini comment: 
We have public API (used for example by RSocket)
`reactor.netty.http.client.HttpClient.WebsocketReceiver#connect`

so the only use case is pure HTTP requests

Reply:
I made `ClientTransport#connect*` methods protected so only the transports that provide direct access to `connect` can expose that functionality.